src/client/ua_client.c

```diff
@@ -20,6 +21,15 @@
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     memset(client, 0, sizeof(UA_Client));
     client->channel.connection = &client->connection;
+    client->channel.endpoint = (UA_Endpoint*)UA_malloc(sizeof(UA_Endpoint));
```

Comment:
The client needs more than one Endpoint though. If we want to connect anonymously we use the standard endpoint like it is used here. But as soon as we want to connect to one of the endpoints of a server that we got from the getEndpoints service, we need to set that endpoint here, so that the algorithms and such that correspond to the endpoint are used. The former can be part of the client, but the latter needs to be dynamically allocated, as soon as we want to connect to that specific endpoint. The latter functionality does not exist yet. (This should probably be part of a new PR)